### PR TITLE
Modify annotation tool can now be used to edit nodes in an annotation item

### PR DESCRIPTION
**Pull request by nyalldawson**
This PR allows the modify annotation tool to change an existing annotation by clicking over an existing node and moving it. Interaction follows the standard QGIS patterns of a left click to start the move, second left click to apply it, or a right-click to cancel.

Selecting a node and pressing delete will remove the node, double clicking a segment will add a node.

https://user-images.githubusercontent.com/1829991/132645562-7dcd3aa2-779f-4f6b-a7bd-13f9e7fa3d3a.mp4

Sponsored by the Swiss QGIS user group.

### CODE REVIEW

File: src/core/maprenderer/qgsrendereditemresults.cpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ class QgsRenderedItemResultsSpatialIndex : public RTree<const QgsRenderedItemDet
 ///@endcond
 
 QgsRenderedItemResults::QgsRenderedItemResults( const QgsRectangle &extent )
-  : mExtent( extent )
+  : mExtent( extent.buffered( std::max( extent.width(), extent.height() ) * 1000 ) )
```
Comment (nirvn) [inline src/core/maprenderer/qgsrendereditemresults.cpp:101]: Can you add a comment here?

File: src/gui/annotations/qgsmaptoolmodifyannotation.cpp [hunk 2]
```diff
@@ -193,22 +204,54 @@ void QgsMapToolModifyAnnotation::cadCanvasMoveEvent( QgsMapMouseEvent *event )
 
     case Action::MoveItem:
     {
+#if 0
```
Comment (nirvn) [inline src/gui/annotations/qgsmaptoolmodifyannotation.cpp:207]: Planning to leave this in or remove?

**Review by nirvn — APPROVED** [review]
Nicely done.

**Comment by nyalldawson** [comment]
Thanks @nirvn!

**Comment by nyalldawson** [comment]
Unrelated test failure

**Pull request merged**